### PR TITLE
Update new_just and to use new pip tools functions

### PR DESCRIPTION
**Pull request by andyneff**
Get the new pip-tools pattern sorted out in new_just. Also adding commonly used just features not previously includes in a new_just, including:

- singular-compose (incomplete)
- deploy
- Unhardcoding vsiri/recipe for recipes. This is useful for docker build caching, deploying, and not conflicting on a shared server

### CODE REVIEW

File: linux/just_files/new_just [hunk 1]
```diff
@@ -585,34 +712,35 @@ function write_dockerfile()
...
             #T# 5) just shell
             #T# 6) python -c "import osgeo.gdal_array" # Should not create an error message
-
-            ARG SKIP_PIPENV_SYNC
-            #T# Sometimes you need to fix the image so that pipenv lock can be updated, but
-            #T# you can'"'"'t because pipenv sync won'"'"'t work. This chicken/egg cycle can be
-            #T# escaped by setting the build arg SKIP_PIPENV_SYNC to 1. Then, pipenv sync is
+            #T#
+            #T# Call pip-fake for each editable package
+            RUN pip-fake "'"$(docker_add_quote_escape "/src/${RELATIVE_PATH}")"'" vsi python/vsi; \
```
[the remark this answers is not in the document]
Comment (andyneff) [reply linux/just_files/new_just:717]: I don't want it first (as that's the `RUN` line) or last (since the last line can't end in `; \`). But putting it in the middle makes is so when it is uncommented, it is syntactically correct. It only shows up with tutorial mode on. So not that big of a deal. I went ahead and changed the order so `/src` comes first.

File: linux/just_files/new_just [hunk 2]
```diff
@@ -585,34 +712,35 @@ function write_dockerfile()
...
-            #T#         python3 -c "import json; print(json.load(open('"'"'${PIPENV_PIPFILE}.lock'"'"', '"'"'r'"'"'))['"'"'default'"'"']['"'"'$1'"'"']['"'"'version'"'"'])" 2>/dev/null || :; \
-            #T#       }; \
-            #T#       # Install numpy first, so that the gdal install works.
-            #T#       pipenv run pip install numpy$(pipfile_lock_version numpy); \
+            #T#       # Add numpy to the prinstall line to install numpy first,
+            #T#       # so that the gdal install works.
+            #T#       pip-8210 install numpy; \
                   # Install all packages into the image
-                  pipenv sync; \
+                  # --no-build-isolation
```
[the remark this answers is not in the document]
Comment (andyneff) [reply linux/just_files/new_just:735]: It was a left over comment. I was trying to set that up, but descoped it from this effort.

File: docker-compose.yml [hunk 3]
```diff
@@ -118,6 +116,11 @@ services:
     image: ${VSI_COMMON_DOCKER_REPO}:os_${VSI_COMMON_TEST_OS_TAG_NAME-debian_8}
     volumes:
       - <<: *vsi_common_volume
+      # Since some of the integration tests bind test dirs in, make sure the
+      # tmp it is using is the same tmp dir
```
Comment (scott-vsi) [inline docker-compose.yml:120]:
```suggestion
      # Because some of the integration tests bind mount test dirs into the
      # container, ensure the tmp dir it is using is the same tmp dir
```

Comment (scott-vsi) [inline docker-compose.yml:120]: why is this necessary?

File: linux/just_files/new_just [hunk 4]
```diff
@@ -485,134 +614,121 @@ function write_dockerfile()
...
+                  #T# requirements.txt as a constraint to install the correct versions of
+                  #T# libraries and their dependencies. Additionally, it handles all
+                  #T# incomptibilies covered in pip issue #8210
+                  pip-8210 install pip-tools; \
+
+                  #T# # Modern pyproject.toml packages shouldn'"'"'t require this, but older setup.py projects
+                  #T# # might have an "build requirment" just to start the build process. Common examples
+                  #T# # of this include numpy and torch. E.g. to install numpy before a package builds,
+                  #T# # add numpy to the pre-install line above to install the correct version of numpy
+                  #T# # like this:
```
Comment (scott-vsi) [inline linux/just_files/new_just:721]:
```suggestion
                  #T# # of this include numpy and torch. E.g. to install the correct version of numpy
                  #T# # before a package builds, add it to the pre-install line above like this
```

Comment (scott-vsi) [reply linux/just_files/new_just:721]: Is this too wide?

File: docker-compose.yml (hunk 3 above)
Comment (andyneff) [reply docker-compose.yml:120]: because a new just project mounts in the source code

**Review by scott-vsi — APPROVED** [review]
